### DISCUSSION TITLE
Build BIRT on mac mini M1 fails

### DISCUSSION THREAD

**Discussion by colinsutton** [eclipse-birt/birt#779] [Q&A]
I installed java 17 for ARM 64  from https://www.oracle.com/java/technologies/downloads/#jdk17-mac 
     java version "17.0.1" 2021-10-19 LTS
    Java(TM) SE Runtime Environment (build 17.0.1+12-LTS-39)
    Java HotSpot(TM) 64-Bit Server VM (build 17.0.1+12-LTS-39, mixed mode, sharing)
After updating my clone of birt-master, eclipse seems to compile OK in my work area:
<img width="1024" alt="screenshot_563" src="https://user-images.githubusercontent.com/18058891/147431990-10149b84-5185-4a14-852f-d688106c47bc.png">

When I select the configurations I see two all-in-one applications:
<img width="1063" alt="screenshot_565" src="https://user-images.githubusercontent.com/18058891/147432002-40894310-db28-4825-ba2b-699e0abe4f73.png">

When I select 'run', the validation complains about every(?) module
<img width="544" alt="screenshot_567" src="https://user-images.githubusercontent.com/18058891/147432051-c87c1eee-9912-4f3f-af3a-0ec084b2392e.png">

When I continue:
<img width="572" alt="screenshot_568" src="https://user-images.githubusercontent.com/18058891/147432067-1173557c-630e-4d66-9995-0a2ff355ebe3.png">

The log shows:
!SESSION Mon Dec 27 12:42:29 AEDT 2021 -----------------------------------------
!ENTRY org.eclipse.equinox.launcher 4 0 2021-12-27 12:42:29.461
!MESSAGE Exception launching the Eclipse Platform:
!STACK
java.lang.ClassNotFoundException: org.eclipse.core.runtime.adaptor.EclipseStarter
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:445)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:587)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:656)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:596)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1467)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1440)

What is wrong?

**Comment by colinsutton** [comment]
On looking into the eclipseIDE 2021-12, supposedly built for AARCH64 I see it is launching using a file labelled [...]x86_64_1.2.400.v20211117-0650  Could this be the wrong architecture?
<img width="760" alt="screenshot_570" src="https://user-images.githubusercontent.com/18058891/147799479-a98620fd-9f61-45d8-b8bd-445303112e73.png">

**Reply by colinsutton** [reply]
Also the session data shows:
eclipse.buildId=4.22.0.I20211124-1800
java.version=11.0.12
java.vendor=Oracle Corporation
BootLoader constants: OS=macosx, ARCH=x86_64, WS=cocoa, NL=en_AU
Framework arguments:  -product org.eclipse.epp.package.committers.product -keyring /Users/colin/.eclipse_keyring
Command-line arguments:  -os macosx -ws cocoa -arch x86_64 -product org.eclipse.epp.package.committers.product -keyring /Users/colin/.eclipse_keyring

Where is this set up? Looks like wrong java, wrong architecture.

**Reply by wimjongman** [reply]
You can copy the launch configuration. In the plugins tab, remove the x86 and add the ones for your platform.

**Reply by wimjongman** [reply]
Did you try this ^^

**Comment by wimjongman** [comment]
We have added aarch64 and have a freshly baked build for you to try here:

https://download.eclipse.org/birt/downloads/drops/snapshot/

**Reply by TheLaserNinja** [reply]
Awesome, I'll test this out this coming week.
